charmcraft.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: oauth2-proxy-k8s
+type: charm
+title: OAuth2 Proxy
+summary: A reverse proxy server that authenticates users through providers like Google and Github.
+description: |
+  OAuth2 Proxy is a reverse proxy and static file server that authenticates 
+  users through providers like Google and GitHub, allowing validation by email, 
+  domain, or group.
+links: 
+  documentation: https://discourse.charmhub.io/t/charmed-oauth2-proxy-k8s-overview/14663
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+# Metadata
+requires:
+  nginx-route:
+    interface: nginx-route
+    limit: 1
+
+# (Optional) Configuration options for the charm
+# This config section defines charm config options, and populates the Configure
+# tab on Charmhub.
+# More information on this section at https://juju.is/docs/sdk/charmcraft-yaml#heading--config
+# General configuration documentation: https://juju.is/docs/sdk/config
+config:
+  options:
+    upstream:
+      description: |
+          The HTTP url(s) of the upstream endpoint. For juju applications, this is 
+          "http://<application_name>" if it is deployed on the same model.
```

Comment:
Not sure if also the port should be included (a juju application may also require exposing/opening the port)